components/automate-cli/pkg/verifyserver/server/api/v1/softwareversion_test.go

```diff
@@ -0,0 +1,98 @@
+package v1_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/server"
+	v1 "github.com/chef/automate/components/automate-cli/pkg/verifyserver/server/api/v1"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/softwareversionservice"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/gofiber/fiber"
+	"github.com/stretchr/testify/assert"
+)
+
+func SetupMockSoftwareVersionService(response models.SoftwareVersionDetails, err error) softwareversionservice.ISoftwareVersionService {
+	return &softwareversionservice.MockSoftwareVersionService{
+		GetSoftwareServicesFunc: func() (*models.SoftwareVersionDetails, error) {
+			return &response, err
+		},
+	}
+}
+
+func SetupHandler(sv softwareversionservice.ISoftwareVersionService) *fiber.App {
+	log, _ := logger.NewLogger("text", "debug")
+	fconf := &fiber.Settings{
+		ServerHeader: server.SERVICE,
+		ErrorHandler: fiberutils.CustomErrorHandler,
+	}
+	app := fiber.New(fconf)
+	handler := v1.NewHandler(log).AddSoftwareVersionService(sv)
+
+	vs := &server.VerifyServer{
+		Port:    server.DEFAULT_PORT,
+		Log:     log,
+		App:     app,
+		Handler: handler,
+	}
+	vs.SetupRoutes()
+	return vs.App
+}
+
+func TestSoftwareVersionAPI(t *testing.T) {
+	tests := []struct {
+		description   string
+		expectedCode  int
+		requestBody   models.SoftwareVersionDetails
+		expectedBody  string
+		expectedError error
+		query         string
+	}{
+		{
+			description:  "200:success software version check route",
```

Comment:
add more test cases for when software are not available, or os versions don't match. Cover all types of responses possible

Reply:
Done